web/src/TestPage.js

```diff
@@ -17,16 +17,20 @@ class TestPage extends React.Component {
       classes: props,
       status: false,
       events: [],
+      eventCount: 0,
       pageLoadTime: new Date(),
       isBackground: false,
       eventBuckets: Array(11).fill(0),
-      eventCount: 0,
       speed: 0,
       payloadSize: 0,
       sessionId: "",
       traces: [],
       trace: null,
     };
+    this.events = [];

```

Comment:
Why not use local variables?

Reply:
If using local variables, you cannot update this.state.events with setState.